jobs/integration/conftest.py

```diff
@@ -143,41 +143,49 @@ def pytest_addoption(parser):
 class Tools:
     """Utility class for accessing juju related tools"""
 
-    def __init__(self, request):
-        self._request = request
+    def __init__(self, config):
+        self._config = config
         self.requests = requests
         self.requests_get = asyncify(requests.get)
+        self.run = asyncify(self.exec)
```

Comment:
the existing `tools.run` is an async version of the new `tools.exec` which does away with the async stuff...